search/views.py

```diff
@@ -105,13 +107,36 @@ def syllabus(request, id):
                   {'course': course, 'syllabus_path': request.path})
 
 
-def hit(request, id):
-    course = get_object_or_404(Course, id=id)
-    course.hit += 1
-    course.save()
+def course_manipulation(request, id):
+    """ Use this `course_manipulation` function to record the course """
+    if request.user.is_authenticated():
+        current_user = Member.objects.get(user=request.user)
+        course = get_object_or_404(Course, id=id)
+        request_type = request.GET.get('type', 'GET')
```

Comment:
did you mean ``request.method``? OAO
[ref] https://docs.djangoproject.com/en/1.7/ref/request-response/#django.http.HttpRequest.method